components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -489,6 +473,94 @@ public void addFederatedAuthenticatorConfig(FederatedAuthenticatorConfig authnCo
         }
     }
 
+    private void updateSingleValuedFederatedConfigProperties(Connection dbConnection, int authnId, int tenantId,
+                                                             List<Property> singleValuedProperties) throws
+            SQLException {
+
+        PreparedStatement prepStmt2 = null;
+        PreparedStatement prepStmt3 = null;
+        String sqlStmt;
+
+        try {
+            for (Property property : singleValuedProperties) {
+
+                sqlStmt = IdPManagementConstants.SQLQueries.UPDATE_IDP_AUTH_PROP_SQL;
+                prepStmt2 = dbConnection.prepareStatement(sqlStmt);
+                prepStmt2.setString(1, property.getValue());
+                if (property.isConfidential()) {
+                    prepStmt2.setString(2, "1");
```

Comment:
Please use constants with a explanatory name so that others can understand the logic behind "0" and "1"